joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/passes/ImportResolverPass.scala

```diff
@@ -28,58 +28,84 @@ class ImportResolverPass(cpg: Cpg, packageTableInfo: PackageTable) extends XImpo
+          .flatMap { typeDeclModel =>
+            Seq(ResolvedMethod(s"${typeDeclModel.fullName}.new", "new"), ResolvedTypeDecl(typeDeclModel.fullName))
+          }
+          .distinct
+
+        val importNodesFromModule = packageTableInfo.getModule(expEntity).flatMap { moduleModel =>
+          Seq(ResolvedTypeDecl(moduleModel.fullName))
+        }
+        (importNodesFromTypeDecl ++ importNodesFromModule).toSet
+      } else {
+        val resolvedTypeDecls = cpg.typeDecl
+          .where(_.file.name(s"${Pattern.quote(expResolvedPath)}\\.?.*"))
+          .fullName
+          .flatMap(fullName => Seq(ResolvedTypeDecl(fullName), ResolvedMethod(s"$fullName.new", "new")))
+          .toSet
+
+        val resolvedModules = cpg.namespaceBlock
+          .whereNot(_.nameExact("<global>"))
+          .where(_.file.name(s"${Pattern.quote(expResolvedPath)}\\.?.*"))
+          .flatMap(module => Seq(ResolvedTypeDecl(module.fullName)))
+          .toSet
+
+        // Expose methods which are directly present in a file, without any module, TypeDecl
+        val resolvedMethods = cpg.method
+          .where(_.file.name(s"${Pattern.quote(expResolvedPath)}\\.?.*"))
+          .where(_.nameExact(":program"))
+          .astChildren
+          .astChildren
+          .isMethod
+          .flatMap(method => Seq(ResolvedMethod(method.fullName, method.name)))
+          .toSet
+        resolvedTypeDecls ++ resolvedModules ++ resolvedMethods
       }
+    }
 
-      val resolvedTypeDecls = cpg.typeDecl
-        .where(_.file.name(s"${Pattern.quote(resolvedPath)}\\.?.*"))
-        .fullName
-        .flatMap(fullName => Seq(ResolvedTypeDecl(fullName), ResolvedMethod(s"$fullName.new", "new")))
-        .toList
-      val resolvedTypeDeclByModule = packageTableInfo
-        .getModulesByGem(resolvedPath)
-        .map(moduleName => ResolvedTypeDecl(s"$entity::program.$moduleName"))
-        .toList
-      val resolvedFunctions = cpg.method
-        .where(_.file.name(s"${Pattern.quote(resolvedPath)}\\.?.*"))
-        .whereNot(_.nameExact(":program"))
-        .map(method => ResolvedMethod(method.fullName, method.name))
-        .toList
-      (resolvedTypeDecls ++ resolvedTypeDeclByModule ++ resolvedFunctions).distinct
-    } else Seq.empty
+    finalResolved
+  }
 
-    val importNodes = methodTableModelList.flatMap { methodTableModel =>
-      Seq(
-        ResolvedMethod(s"$expEntity::program.${methodTableModel.parentClassPath.stripSuffix(":")}.new", "new"),
-        ResolvedTypeDecl(s"$expEntity::program.${methodTableModel.parentClassPath}")
+  def getResolvedPath(expEntity: String, fileName: String) = {
+    val rawEntity   = expEntity.stripPrefix("./")
+    val matcher     = pathPattern.matcher(rawEntity)
+    val sep         = Matcher.quoteReplacement(JFile.separator)
+    val root        = s"$codeRoot${JFile.separator}"
+    val currentFile = s"$root$fileName"
+    val entity      = if (matcher.find()) matcher.group(1) else rawEntity
```

Comment:
If `matcher.find()` is true, do we always know that `group(1)` will be available?

Reply:
Yes, Also this has been referred from Javascript